### PR TITLE
Use tmp particle container instead of WritePlotFile with filter

### PR DESCRIPTION
**Pull request by atmyers**

### CODE REVIEW

File: Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp [hunk 1]
```diff
@@ -309,18 +313,22 @@ FlushFormatPlotfile::WriteParticles(const std::string& dir,
...
+
+        using SrcData = WarpXParticleContainer::ParticleTileType::ConstParticleTileDataType;
+        tmp.copyParticles(*pc,
+        [=] AMREX_GPU_HOST_DEVICE (const SrcData& src, int ip, const amrex::RandomEngine& engine)
+        {
+            const SuperParticleType& p = src.getSuperParticle(ip);
+            return random_filter(p, engine) * uniform_filter(p, engine)
+                * parser_filter(p, engine) * geometry_filter(p, engine);
+        }, true);
+
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp:325]: @RemiLehe asks if we should sync here?

File: Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp [hunk 2]
```diff
@@ -309,18 +313,22 @@ FlushFormatPlotfile::WriteParticles(const std::string& dir,
...
         // real_names contains a list of all particle attributes.
         // particle_diags[i].plot_flags is 1 or 0, whether quantity is dumped or not.
-        pc->WritePlotFile(
+        tmp.WritePlotFile(
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp:328]: I fear we need to `printf()` debug the old and new routines

**Review by ax3l — APPROVED** [review]
Wuhu!

**Comment by RevathiJambunathan** [comment]
yayyyyyyy!!!! Thank you @atmyers

**Pull request merged**